### PR TITLE
Incorporated Selenium Page Object Model into Testing

### LINKED ISSUE #1312: Incorporate Selenium Page Object Model For Making Test Cases
Selenium page object model helps us in making more readable, maintainable and extendable code. It eliminates the redundancy and promotes better reuse of code. In a nutshell, Page Object Model is a design pattern to create Object Repository for web UI elements. Under this model, for each web page in the application, there should be corresponding page class.This Page class will find the WebElements of that web page and also contains Page methods which perform operations on those WebElements.

More detailed explanation can be found [here](http://www.guru99.com/page-object-model-pom-page-factory-in-selenium-ultimate-guide.html).

I will send a PR soon incorporating these changes. Thanks!!

### PR DESCRIPTION
**Pull request by Princu7**
Fixes #1312 

Changes:
* Implements Selenium Page Object model for better structure, readability, maintainability of functional tests written for the website.
* No additional tests are written. Just the existing one has been modified to follow the model.

Sample Test Video: The video is short and not interesting due to the lesser number of tests currently. It will become better with added tests.
https://saucelabs.com/beta/tests/caa07dfbfca94e7d80cb2100742d4daf/watch#6

@aayusharora @geekyd Please review. Thanks!!

### CODE REVIEW

**Comment by mariobehling** [comment]
@Princu7 Please actively ask your co-developers to review using the "@" feature and ping them on chat if necessary. Help to speed up the review and merge process.

Please review: @CosmicCoder96 @anu-007 @geekyd @harshitagupta30 @sumedh123 @utkarshnath

File: test/serverTest.js [hunk 1]
```diff
@@ -188,13 +191,20 @@ describe("Running Selenium Tests on Chrome Driver", function() {
...
+  describe('Testing event page', function() {
+    var eventPage;
 
-    driver.get("http://localhost:5000/live/preview/test@event.com/FOSSASIA%202016");
-    var headline = driver.findElement(webdriver.By.css('h1'));
-    headline.getText().then(function(txt) {
-      assert.equal(txt, "FOSSASIA 2016");
-      done();
+    before(function() {
+      eventPage = new EventPage(driver);
```
Comment (geekyd) [inline test/serverTest.js]: Great job!

File: src/selenium/basePage.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+var until = require('selenium-webdriver').until;
+var By = require('selenium-webdriver').By;
+
+function BasePage(webdriver) {
+  this.driver = webdriver;
+}
+
+BasePage.prototype.visit = function(url) {
```
Comment (geekyd) [inline src/selenium/basePage.js]: This is good but this is ES5 standard, we should follow ES6 standards.

Comment (Princu7) [inline src/selenium/basePage.js]: @geekyd Thanks for the review :). You are right. I used the hackish trick of emulating class behavior in ES5. I then read about the class introduced in the ES6 and they are nothing but an abstraction over the actual prototypical inheritance Javascript provides. I just revised about the prototypical inheritance and its advantage over the class inheritance. There is no tight coupling and inflexible hierarchy problem. Also Class doesn't really exist in JS and they were just introduced for syntactic sugar and to reduce the wrong hackish behavior to emulate it (like I did :grinning:) and provide that as a built-in feature. 

So, I redid the thing using prototypical inheritance here. The code is much cleaner now. Please review again. Thanks!!

**Comment by Princu7** [comment]
@geekyd Thanks for the review :). You are right. I used the hackish trick of emulating class behavior in ES5. I then read about the class introduced in the ES6 and they are nothing but an abstraction over the actual prototypical inheritance Javascript provides. I just revised about the prototypical inheritance and its advantage over the class inheritance. There is no tight coupling and inflexible hierarchy problem. Also Class doesn't really exist in JS and they were just introduced for syntactic sugar and to reduce the wrong hackish behavior to emulate it (like I did :grinning:) and provide that as a built-in feature. 

So, I redid the thing using prototypical inheritance here. The code is much cleaner now. Please review again. Thanks!!

File: src/selenium/basePage.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+var until = require('selenium-webdriver').until;
+var By = require('selenium-webdriver').By;
+
+var BasePage = {
```
Comment (geekyd) [inline src/selenium/basePage.js]: @Princu7 Good Job!

**Pull request merged**